### PR TITLE
feat: implement sidebar topic list

### LINKED ISSUE #419: [Product Topic] implement sidebar topic list

### PR DESCRIPTION
**Pull request by duynguyen**
Fix #419 

Test URLs:
- Before: https://main--danaher-ls-aem--hlxsites.hlx.page/us/en/products/assay-kits/topics/what-are-assay-kits
- After: https://419--danaher-ls-aem--hlxsites.hlx.page/us/en/products/assay-kits/topics/what-are-assay-kits

### CODE REVIEW

**Review by davenichols-DHLS — APPROVED** [review]
I love this, thank you!

**Review by davenichols-DHLS — CHANGES_REQUESTED** [review]
Thank you @duynguyen , this is fantastic, but sorry, can I ask for a few changes please?

- Hide left bar on mobile view, like we do with the right bar on blogs/news
- Can we add a hover over of purple-25 (sorry this was missed on the UX)
- Reduce the left hand bar to 25% of width

**Comment by sharanagoudapatil** [comment]
@duynguyen - Please verify for these 2 pages, social-media links still not working as expected
 
https://419--danaher-ls-aem--hlxsites.hlx.page/us/en/products/assay-kits/topics/how-are-assay-kits-installed 
https://419--danaher-ls-aem--hlxsites.hlx.page/us/en/products/assay-kits/topics/what-are-assay-kits 

image

**Review by mhaack — APPROVED** [review]

File: blocks/side-nav/side-nav.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import ffetch from '../../scripts/ffetch.js';
+import { getMetadata } from '../../scripts/lib-franklin.js';
+import { makePublicUrl } from '../../scripts/scripts.js';
+import { fetchTopicsForCategory } from '../topic-list/topic-list.js';
+
+async function fetchAllProductCategories() {
+  const products = await ffetch('/us/en/products-index.json')
+    .filter(({ type, path }) => type === 'Category' && path.endsWith('/topics'))
+    .all();
+  return products.sort((item1, item2) => item1.title.localeCompare(item2.title));
```
Comment (mhaack) [inline blocks/side-nav/side-nav.js:11]: Better rename to "category" as these are not products.

Comment (duynguyen) [reply blocks/side-nav/side-nav.js]: done

File: blocks/side-nav/side-nav.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { div, a, strong } from '../../scripts/dom-builder.js';
+import ffetch from '../../scripts/ffetch.js';
+import { getMetadata } from '../../scripts/lib-franklin.js';
+import { makePublicUrl } from '../../scripts/scripts.js';
+import { fetchTopicsForCategory } from '../topic-list/topic-list.js';
+
+async function fetchAllProductCategories() {
+  const topicHubs = await ffetch('/us/en/products-index.json')
+    .filter(({ type, path }) => type === 'TopicHub' && path.endsWith('/topics'))
```
Comment (mhaack) [inline blocks/side-nav/side-nav.js:9]: The check `path.endsWith('/topics')` is not needed any more right?

Comment (duynguyen) [reply blocks/side-nav/side-nav.js]: fixed, thanks

**Pull request merged**